### ISSUE TITLE
Army Controllers

### ISSUE THREAD

**Issue by iii235** [DFHack/dfhack#1781]
Do you guys know enough about the controllers to remake the force siege script, so that it redirects an army to your fortress or forces a civ to send an army?

**Comment by iii235** [comment]
This is what I get when I run it with the spaces

"...s\lazynewb\Dwarf Fortress 0.47.05/hack/scripts/s-gen.lua:125: attempt to index a nil value (field 'T_unk_2c')
stack traceback:
        ...s\lazynewb\Dwarf Fortress 0.47.05/hack/scripts/s-gen.lua:125: in local 'script_code'
        ...oads\lazynewb\Dwarf Fortress 0.47.05\hack\lua\dfhack.lua:680: in function 'dfhack.run_script_with_env'
        (...tail calls...)"

**Comment by iii235** [comment]
Can't I just accomplish this with the gm editor?

**Comment by lethosor** [comment]
From the error message, I would guess that `T_unk_2c` was renamed to something else since the script was written. There could be other issues - this is just the first one that comes up. Looks like some other fields used in the script were renamed in https://github.com/DFHack/df-structures/commit/96f1474f5e68737fb4719513c85adac33f436c54, for instance.

From looking at the script, it looks like it would be rather complicated to do through `gui/gm-editor`, if possible at all.

**Comment by warmist** [comment]
I'll try to update it. Though the [These lines](https://gist.github.com/pronvit/e256267e3c65c0e89e08b1aa947df600#file-s-gen-lua-L186) are both funny and sad at the same time :D

**Comment by warmist** [comment]
Update: made it work*. Newer version: [here](https://gist.github.com/warmist/2c3d50c265b14ec60c749a04fbdc0132)

*: for some values of work -

-  will probably corrupt your world!
- said "undead are walking" when humans spawned?!
- second command did nothing while the first siege was active (or just did not work?)
- Either way more testing/work needed on the script. Probably most of the fields can be not filled out. Most important (IMHO) is the data field and squads and all the references/ids

**Comment by iii235** [comment]
Does it work in fortress mode?

**Comment by lethosor** [comment]
Why do you ask? I haven't tried it, but given that it relies on a couple fortress mode fields being set (ui.civ_id, ui.site_id), I don't think it could work in any mode *except* fortress mode.

**Comment by warmist** [comment]
Yup that is correct: tested only in fortress mode. Imho next steps would be to cut down everything that is not needed, try to figure out the needed ones names and add to df-structures, release new dfhack with the script/structure changes.
